frontend/src/metabase/entities/schemas.js

```diff
@@ -25,13 +25,21 @@ export default createEntity({
   name: "schemas",
   schema: SchemaSchema,
   api: {
-    list: async ({ dbId, getAll = false, ...args }) => {
+    list: async ({ dbId, getAll = false, ...args }, dispatch) => {
       if (!dbId) {
         throw new Error("Schemas can only be listed for a particular dbId");
       }
-      const schemaNames = await (getAll
-        ? MetabaseApi.db_syncable_schemas({ dbId, ...args }) // includes empty schema
-        : MetabaseApi.db_schemas({ dbId, ...args }));
+      const schemaNames = getAll
```

Comment:
new code is not clear in terms of what goes to if and what goes to else and why there is such distinguish